src/gfx/shader/wgsl_mapping.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once

```

Comment:
can we have a readme.md doc in this folder explaining how this whole thing works btw.

Also pragma once is not really the best (but should be ok for now), prefer proper define guards as they support all corner cases

Reply:
Will create a small document for the shader tree/generator (https://github.com/orgs/fragcolor-xyz/projects/1#card-79476718)

Reply:
Made a note about header guards so I can work on it later and don't forget. (https://github.com/orgs/fragcolor-xyz/projects/1#card-79476639)